openedx/stanford/djangoapps/auth_lagunita/tests/test_api.py

```diff
@@ -0,0 +1,259 @@
+        response = MockedResponse(200, json_dumps(DATA['exists']))
+        with patch('requests.post', return_value=response):
+            result, error = self.api._create_lead(self.user)
+            self.assertIsNone(result)
+            self.assertIsNotNone(error)
+            self.assertEquals(error, 'skipped')
+
+    def test_error(self):
+        response = MockedResponse(200, json_dumps(DATA['error']))
+        with patch('requests.post', return_value=response):
+            result, error = self.api._create_lead(self.user)
+            self.assertIsNone(result)
+            self.assertIsNotNone(error)
+
+    def test_exception(self):
+        with patch('requests.post', side_effect=requests.exceptions.HTTPError):
+            result, error = self.api._create_lead(self.user)
+            self.assertIsNone(result)
+            self.assertIsNotNone(error)
+
+
+class MarketoApiLookupTest(TestCase):
+
+    def setUp(self):
+        super(MarketoApiLookupTest, self).setUp()
+        self.api = MarketoApi(
+            'localhost',
+            access_token='foobar',
+        )
+        self.api.refresh_access_token = Mock(return_value='dead-beef-cafe')
+        self.user = MockedUser()
+
+    def test_auth_token(self):
+        access_token = self.api.refresh_access_token()
+        self.assertEquals(access_token, 'dead-beef-cafe')
+
+    def test_empty(self):
+        response = MockedResponse(200, '')
+        with patch('requests.get', return_value=response):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNone(lead)
+
+    def test_error_http_code(self):
+        response = MockedResponse(404, '')
+        with patch('requests.get', return_value=response):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNone(lead)
+
+    def test_error_http_exception(self):
+        with patch('requests.get', side_effect=requests.exceptions.HTTPError):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNone(lead)
+
+    def test_error(self):
+        response = MockedResponse(200, json_dumps(DATA['error']))
+        with patch('requests.get', return_value=response):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNone(lead)
+
+    def test_nonexisting(self):
+        data = {
+            'success': True,
+            'errors': [
+            ],
+            'result': [
+            ],
+        }
+        response = MockedResponse(200, json_dumps(data))
+        with patch('requests.get', return_value=response):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNone(lead)
+
+    def test_existing(self):
+        response = MockedResponse(200, json_dumps(DATA['success']))
+        with patch('requests.get', return_value=response):
+            lead = self.api.lookup_lead(self.user.email)
+            self.assertIsNotNone(lead)
+
+    def test_retry_auth_token(self):
+        pass
```

Comment:
Sorry, what's this for?

Reply:
This was a placeholder;
I have now added a relevant test to ensure that:
- if a request is attempted with an expired/invalid auth_token,
  - it will refresh the token and try the original request one more time

This is to handle the case where the command is long-running and the auth token expires.